src/framework-ondata.js

```diff
@@ -308,7 +308,11 @@ function handleOutput(vni, port, lastLm, state ) {
             port.disconnect();
 
             if (nodeInstance.isTransient) { 
-                vni.delete();
+                if (vni.vnid === '') { 
+                    vni.clearTransientInputs();
```

Comment:
This behaviour is very subtle and could use a comments like:
* Clear empty-string vni states, that are not single IIPs, when only empty-string vni states are used as input